dftimewolf/metawolf/utils.py

```diff
@@ -0,0 +1,361 @@
+              settable) if unmarshal else settable
+
+    return loaded_sessions
+
+  def RunInBackground(
+      self,
+      processes: List['output.MetawolfProcess']
+  ) -> bool:
+    """Guard executed before `quit` or SIGINT.
+
+    If processes are still running, ask the user if they want to let them run
+    in the background.
+
+    Args:
+      processes (List[output.MetawolfProcess]): The processes associated to
+          Metawolf's session.
+
+    Returns:
+      bool: True if the processes should keep running in the background.
+    """
+    still_running = False
+    for metawolf_process in processes:
+      if metawolf_process.Poll() is None:
+        still_running = True
+        break
+    if still_running:
+      value = input('Metawolf still has running processes. Would you like to '
+                    'keep them running in the background [Yn]? ') or 'y'
+      q = self.Str2Bool(str(value))
+      while q not in [False, True]:
+        value = input('[Yn]? ') or 'y'
+        q = self.Str2Bool(str(value))
+      return q
+    return True
+
+  def PrepareDFTimewolfCommand(
+      self,
+      recipe: str,
+      session_settables: Dict[str, session.SessionSettable]
+  ) -> List[str]:
+    """Builds a cli command from current session parameters.
+
+    Args:
+      recipe (str): The name of the recipe to run.
+      session_settables (Dict[str, SessionSettable]): The dictionary holding the
+          session settables to use to build the command.
+
+    Returns:
+      List[str]: A list that contains the CLI command elements based on the
+          current recipe and settables.
+    """
+    cmd_components = {}
+    for _, settable in session_settables.items():
+      if settable.name == SESSION_ID_SETTABLE:
+        continue
+      if not settable.IsOptional():
+        value = settable.GetValue()
+        if value is None:
+          return []
+        cmd_components[settable.name] = '{0!s}'.format(value)
+      else:
+        value = settable.GetValue()
+        if value:
+          cmd_components[settable.name] = '--{0:s}={1!s}'.format(
+              settable.name, value)
+
+    # We need to order the arguments as they are defined in the recipe
+    cmd = [DFTIMEWOLF, recipe]
+    for name, _, _ in self.recipe_manager.Recipes()[recipe].args:
+      if name.startswith('--'):
+        name = name.replace('--', '')
+      if name in cmd_components:
+        cmd.append(cmd_components[name])
+    return cmd
+
+  def GetRecipes(self) -> Dict[str, str]:
+    """Return available DFTimewolf recipes.
+
+    Returns:
+      Dict[str, str]: A dictionary that maps recipe names to their short
+          description.
+    """
+    recipes = {}
+    for recipe in self.recipe_manager.GetRecipes():
+      short_desc = ' '.join(recipe.GetHelpString().split())
+      short_desc = short_desc.replace(recipe.name, '').strip()
+      recipes[recipe.name] = short_desc
+    return recipes
```

Comment:
Maybe you could just access `recipe.contents['short_description']` instead? 

```python
short_desc = recipe.contents.get('short_description', 'No description')
```

Reply:
Or even better, add a `short_description` property to resources.Recipe that does the dict checking for you, and adjust `GetHelpString` accordingly

Reply:
IIRC I didn't access it directly because it is formatted in a weird way (to work with argparser help). It contains several spaces next to one another, which is what I'm removing l343. It also contains the recipe name, which I'm not interested in as part of the description, and that I remove l344. What do you think?

Reply:
```
>>> tool._recipes_manager._recipes['gcp_forensics'].contents['short_description']
'Copies disk from a GCP project to an analysis VM.'
```

This is just the content of the `short_description` field in the recipe JSON files. Where is it formatted weirdly?

Reply:
Ah, I was referring to the output of `GetHelpString()`; I read that too fast. Changed to directly fetching the short desc.